src/hooks/useOutputMaterial.ts

```diff
@@ -59,39 +67,25 @@ export function useOutputMaterial(sessionId: string, outputIdOrName: string, mat
 		// get all geometry and materials below the node
 		const geometryData = getGeometryData(node);
 
-		let MaterialClassType: typeof MaterialStandardData | typeof MaterialSpecularGlossinessData | typeof MaterialUnlitData | typeof MaterialGemData;
+		let newMaterial: MaterialStandardData | MaterialSpecularGlossinessData | MaterialUnlitData | MaterialGemData;
 		switch (materialType) {
 		case MaterialType.SpecularGlossiness:
-			MaterialClassType = MaterialSpecularGlossinessData;
+			newMaterial = new MaterialSpecularGlossinessData(materialProps);
 			break;
 		case MaterialType.Unlit:
-			MaterialClassType = MaterialUnlitData;
+			newMaterial = new MaterialUnlitData(materialProps);
 			break;
 		case MaterialType.Gem:
-			MaterialClassType = MaterialGemData;
+			newMaterial = new MaterialGemData(materialProps);
 			break;
 		default:
-			MaterialClassType = MaterialStandardData;
+			newMaterial = new MaterialStandardData(materialProps);
 		}
 
-		let newMaterial: MaterialStandardData | MaterialSpecularGlossinessData | MaterialUnlitData | MaterialGemData | undefined;
 		// update all geometry materials
 		geometryData.forEach(data => {
-			if (data.material && data.material instanceof MaterialClassType) {
-				// update existing material
-				for ( const p in materialProps)
-					(<any>data.material)[p as keyof typeof MaterialClassType] 
-					= materialProps[p as keyof typeof materialProps];
-
-				data.material.updateVersion();
-			} else {
-				// if we didn't already create a new material, do it now
-				if (!newMaterial)
-					newMaterial = new MaterialClassType(materialProps);
-
-				// no material found: define new material
-				data.material = newMaterial;
-			}
+			originalGeometryAndMaterialAssignment.push({ geometry: data, material: data.material });
```

Comment:
@MajorMeerkatThe3rd in case the hook gets used for different values of `sessionId` or `outputIdOrName`, this would fail. We need to keep this list in an object keyed by a combination of the two.

Reply:
@snabela Understood, I created a new commit and changed it to a dictionary with the id of the the geometry. I didn't use a combination of the two, as the material can also be null.

Reply:
@snabela see commit here: 90a58403d57eee089fa9f6da2bb9fe3543ea1944

Reply:
@MajorMeerkatThe3rd I am testing and debugging the hook and need to do some fixes. A question: Does every `IGeometryData` have a unique id? If so, I would remember this one in the state.

Reply:
@snabela Yes, every GeometryData has a unique ID, BUT the GeometryData can be re-used. It shouldn't matter in this case, just wanted to mention it